JoomGalleryfriends/JG4-dev
# Improved messaging trait

## What is this trait for

The messaging trait adds a session based message handling to the component class (Joomgallery\Component\Joomgallery\Administrator\Extension\JoomgalleryComponent).
It allows to collect warnings, debug information and error and output them as one message even over multiple reloads of the page. This is needed for operations with a long runtime needing multiple site refreshes to complete.
In JG3 there was the debugoutput which acted in a similar way. This behaviour will be implemented with that in a more robust and expandable way.

## How to test this PR
Create new images and categories. The behavior and the printed messages should not change from before this PR.
